ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/TransportToInternal.java

```diff
@@ -0,0 +1,452 @@
+                    break;
+                case RespConstants.HYPOGLYKEMI_FORMAGA_KANNA_VARNINGSTECKEN_SVAR_ID:
+                    handleHypoglykemiFormagaKannaVarningstecken(hypoglykemi, svar);
+                    break;
+                case RespConstants.HYPOGLYKEMI_VIDTA_ADEKVATA_ATGARDER_SVAR_ID:
+                    handleHypoglykemiVidtaAdekvataAtgarder(hypoglykemi, svar);
+                    break;
+                case RespConstants.HYPOGLYKEMI_ATERKOMMANDE_SENASTE_ARET_SVAR_ID:
+                    handleHypoglykemiAterkommandeSenasteAret(hypoglykemi, svar);
+                    break;
+                case RespConstants.HYPOGLYKEMI_ATERKOMMANDE_VAKET_SENASTE_TOLV_SVAR_ID:
+                    handleHypoglykemiAterkommandeVaketSenasteTolv(hypoglykemi, svar);
+                    break;
+                case RespConstants.HYPOGLYKEMI_ALLVARLIG_SENASTE_TOLV_MANADERNA_SVAR_ID:
+                    handleHypoglykemiAllvarligSenasteTolvManaderna(hypoglykemi, svar);
+                    break;
+                case RespConstants.HYPOGLYKEMI_REGELBUNDNA_BLODSOCKERKONTROLLER_SVAR_ID:
+                    handleHypoglykemiRegelbundnaBlodsockerkontroller(hypoglykemi, svar);
+                    break;
+
+                case RespConstants.OVRIGT_KOMPLIKATIONER_AV_SJUKDOMEN_SVAR_ID:
+                    handleKomplikationerAvSjukdomen(ovrigt, svar);
+                    break;
+                case RespConstants.OVRIGT_BOR_UNDERSOKAS_AV_SPECIALIST_SVAR_ID:
+                    handleBorUndersokasAvSpecialist(ovrigt, svar);
+                    break;
+
+                case RespConstants.BEDOMNING_SVAR_ID:
+                    handleBedomning(bedomningUppfyllerBehorighetskrav, svar);
+                    break;
+                case RespConstants.BEDOMNING_OVRIGA_KOMMENTARER_SVAR_ID:
+                    handleOvrigaKommentarer(bedomning, svar);
+                    break;
+                 default:
+                    break;
+            }
+        }
+
+        if (!intygAvserSet.isEmpty()) {
+            utlatande.setIntygAvser(IntygAvser.create(intygAvserSet));
+        }
+        utlatande.setAllmant(allmant.build());
+        if (!bedomningUppfyllerBehorighetskrav.isEmpty()) {
+            bedomning.setUppfyllerBehorighetskrav(bedomningUppfyllerBehorighetskrav);
+        }
+        utlatande.setBedomning(bedomning.build());
+        utlatande.setHypoglykemi(hypoglykemi.build());
+        utlatande.setOvrigt(ovrigt.build());
+    }
+
+    private static void handleIntygAvser(Set<IntygAvserKategori> intygAvserSet, Svar svar) throws ConverterException {
+        for (Delsvar delsvar : svar.getDelsvar()) {
+            switch (delsvar.getId()) {
+                case RespConstants.INTYGETAVSER_DELSVAR_ID:
+                    intygAvserSet.add(IntygAvserKategori.valueOf(getCVSvarContent(delsvar).getCode()));
+                    break;
+                default:
+                    throw new IllegalArgumentException();
```

Comment:
This comment is for all throws of IllegalArgumentException. It would be useful if the value that is illegal was set as message for the exception. If the exception would be thrown one day it will be simplify the work to find what was wrong.

Reply:
fixat